src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -83,6 +88,34 @@ def sql(self) -> str:
     def __str__(self) -> str:
         return self.pretty_name
 
+    @property
+    def plan_node_category(self) -> PlanNodeCategory:
+        return PlanNodeCategory.OTHERS
+
+    @property
+    def individual_node_complexity(self) -> Dict[PlanNodeCategory, int]:
+        """Returns the individual contribution of the expression node towards the overall
+        compilation complexity of the generated sql.
+        """
+        return {self.plan_node_category: 1}
+
+    @property
+    def cumulative_node_complexity(self) -> Dict[PlanNodeCategory, int]:
+        """Returns the aggregate sum complexity statistic from the subtree rooted at this
+        expression node. Statistic of current node is included in the final aggregate.
```

Comment:
add a comment here to mention that the node complexity is the sum of its individual complexity and children complicity, please make sure overwrite the individual_node_complexity properly for new nodes to get the cumulative complexity correct.

Reply:
added